rust/core-lib/src/movement.rs

```diff
@@ -88,14 +84,84 @@ fn vertical_motion(
     } else {
         line.saturating_add(line_delta as usize)
     };
-    let n_lines = view.line_of_offset(text, text.len());
     if line > n_lines {
         return (text.len(), Some(col));
     }
     let new_offset = view.line_col_to_offset(text, line, col);
     (new_offset, Some(col))
 }
 
+/// Compute movement based on vertical motion by the given number of lines skipping
+/// any line that is shorter than the current cursor position.
+fn vertical_motion_exact_pos(
+    r: SelRegion,
+    view: &View,
+    text: &Rope,
+    line_delta: isize,
```

Comment:
I'm not sure that `line_delta` really makes sense as an argument in this case? we're moving up and down by a unit, and that unit is a function of the text/view and the current region, so this could just be an `is_down` bool or something? This saves us having to worry about handling negative numbers, integer conversion, etc.

Reply:
My thought was that we could do a move where line_delta would be a minimum, but I'm not sure how much sense that makes. I will change it.